### ISSUE TITLE
SIRIUS: wrong type of electronic structure method: FULL_POTENTIAL_LAPWLO

### ISSUE THREAD

**Issue by dev-zero** [cp2k/cp2k#1755]
In version 8.2 the following worked:

```
    &PARAMETER
      ELECTRONIC_STRUCTURE_METHOD FULL_POTENTIAL_LAPWLO
    ...
```

with SIRIUS 7.2.7 and CP2K 4ef9989 I get now:

```
SIRIUS: run-time error

=== Fatal error at line 236 of file /tmp/software/spack-stage/spack-stage-sirius-7.2.7-2onvcagphket6ecprltlo67h3ywpxue7/spack-src/src/context/simulation_parameters.cpp ===
wrong type of electronic structure method: FULL_POTENTIAL_LAPWLO
```

and have to correct to

```
    &PARAMETER
      ELECTRONIC_STRUCTURE_METHOD full_potential_lapwlo
    ...
```

... and the manual doesn't list allowed values anymore.

The second issue might already be fixed with one of the commits of @mtaillefumier in SIRIUS already.
Would be good to have a new SIRIUS release before the next CP2K release.

**Comment by mtaillefumier** [comment]
We are cleaning up the api and these problems will be solved at the same time. hopefully before the next release of CP2K. I have the patch ready for the api modifications in my pipeline. 

Note that SIRIUS options **are** case sensitive normally while cp2k options are not. The problem comes from this not SIRIUS specifically. SIRIUS converts the section and options name to lower case internally but does not do anything to the option value.

**Comment by dev-zero** [comment]
There are more inconsistencies. For example when setting the solver to `anderson` it still lists `davidson` as the selected one.

**Comment by dev-zero** [comment]
also, `MPI_GRID_DIMS` seems to be misdocumented: afaik that should be an `int[2]`, not an `int[3]`?

**Comment by dev-zero** [comment]
The main issue with the capitalization of string parameters has been solved in #1784.
The `MPI_GRID_DIMS` is still listed as `<N_VAR>-1</N_VAR>` rather than `<N_VAR>2</N_VAR>`
